app/src/org/commcare/activities/EntitySelectSearchUI.java

```diff
@@ -56,14 +52,13 @@ private void initUIComponents() {
 
     protected CommCareActivity.ActionBarInstantiator getActionBarInstantiator() {
         // again, this should be unnecessary...
-        return (searchItem, searchView, barcodeItem) -> {
-            EntitySelectSearchUI.this.searchMenuItem = searchItem;
-            EntitySelectSearchUI.this.searchView = searchView;
-            EntitySelectSearchUI.this.barcodeMenuItem = barcodeItem;
-
+        return (searchItem, view, barcodeItem) -> {
+            searchMenuItem = searchItem;
+            searchView = view;
+            barcodeMenuItem = barcodeItem;
             restoreLastQuery();
```

Comment:
Looks like this method already restores the search after rotation. Is it not working as intended ?

Reply:
Ohh, I overlooked that. I'm not sure how it's working right now, but I'll check it out.

Reply:
I checked it and seems like `restoreLastQuery` handles rotation only when entity select activity is on top. But when user opens  scanner and then rotate the screen, it won't work.

Reply:
And what's the reason behind that. Last query seems to be saved as part of CC session so it should not get cleared right ?